### PR TITLE
[OLD] Creating JS REST interface for indexer

### PR DESCRIPTION
**Pull request by hariria**
## Motivation

> A Typescript interface for querying the Aptos Postgres Indexer DB

This is an express.js based REST client that interacts with [Prisma](https://www.prisma.io/) as an ORM for querying
the Aptos Postgres Indexer DB.

We are using the Rust -> Diesel models as a source of truth for the database schema. 
The database schema that Prisma uses is defined in the `indexer > typescript > prisma > schema.prisma` file.

In the future, a lot of this code can be reused for a typescript `Aptos Indexer Client` in the future

Comes complete with:
- [x] prisma setup
- [x] testing (coming soon, but the framework is there for jest)
- [x] linting

Please follow the steps in the README.md to try it out

### Major Changes

Modified Indexer folder structure

Before:
```bash
> indexer
  > migrations
  > src
  > etc
```

Now:
```bash
> indexer
  > rust
    > migrations
    > src
  > typescript
    > src
```

## Test Plan

https://user-images.githubusercontent.com/31301117/169383152-56df0431-74da-4825-b285-8bffa90ab077.mov

### CODE REVIEW

File: ecosystem/indexer/rust/README.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ one note: *transaction processing is guaranteed at least once*. It's possible fo
 receive the same transaction more than once: and so your implementation must be idempotent.
 
 To implement your own `TransactionProcessor`, check out the documentation and source code
-here: [`./src/indexer/transaction_processor.rs`](./src/indexer/transaction_processor.rs).
+here: [`./src/indexer/transaction_processor.rs`](srcndexer/transaction_processor.rs).
```
Comment (areshand) [inline ecosystem/indexer/rust/README.md:111]: is this a typo?

**Comment by CapCap** [comment]
why server in ts vs rust? any time indexer changes (i.e diesel changes), we have to update prism schema now too, whereas with rust we can re-use all of the existing models (which are already fully json_serde serializable :-))

File: ecosystem/indexer/rust/README.md (hunk 1 above)
Comment (hariria) [reply ecosystem/indexer/rust/README.md]: fixed

**Comment by hariria** [comment]
@CapCap I agree, you bring up valid points. The primary reason was because we thought we could develop it quicker in TS. Also we can export TS response / request types for the TS SDK. Updating Prisma is a one line command so it's not a huge issue and the TS stuff is already json serializable. I can also just do the warp / actix-web backend as well, shouldn't be too bad :D.

**Pull request closed without merging**